docs/development/malware-checks.rst

```diff
@@ -0,0 +1,109 @@
+
+* The check provides a low-quality or noisy signal (e.g. many false positives), and should be removed. At this point, the check will be moved into a `wiped_out` state, removing all verdicts generated by the evaluation run, and the code for the check will be removed in the next release.
+* The check provides a useful signal, but requires modifications. The administrator will request changes in the initial issue.
+* The check provides a useful signal, and the administrator enables it.
+
+Adding New Checks
+~~~~~~~~~~~~~~~~~
+All active checks are defined as classes in the `warehouse/malware/checks/`_ directory, and exported from ``__init__.py``. The checks in `tests/common/checks/`_ can serve as templates for developing new checks. Simply copy/paste the desired check template into `warehouse/malware/checks/`_ and edit the `dunder init file`_ to get started.
+
+All malware check classes should inherit from ``warehouse.malware.checks.base.MalwareCheckBase``, define a `scan` method, and set the following fields as class attributes:
+
+* ``version`` - ``1`` for new checks, incrementing by one with every subsequent change
+* ``short_description`` - a terse description of the check's purpose
+* ``long_description`` - a more detailed rationale for the check
+* ``check_type`` - either ``event_hook`` or ``scheduled``
+
+For each check type, there is an additional required attribute:
+
+* ``hooked_object``- only for ``event_hook`` checks. The name of the object whose creation triggers a check run. Currently ``"File"``, ``"Release"``, and ``"Project"`` are supported.
+* ``schedule`` - only for ``scheduled`` checks. This should be represented as a dictionary that is passed to a `celery crontab`_.
+
+In ``event_hook`` checks, the id of the ``hooked_object`` will be passed to the ``scan`` method as ``obj_id`` within ``**kwargs``, and should be propagated to the verdict. The `MalwareVerdict model`_ contains more information about required and optional verdict fields.
+
+
+Modifying Existing Checks
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Every time that the code for an existing check is modified, the developer should increment the check version number. This is to ensure that each verdict is associated with a particular version of a check.
+
+Workflow and Testing
+~~~~~~~~~~~~~~~~~~~~
+There are a few steps for executing new malware checks on a development environment:
+
+#. Open `dev/environment`_ and set ``MALWARE_CHECK_BACKEND=warehouse.malware.services.DatabaseMalwareCheckService``. In the development environment, warehouse will, by default, only print the name of the queued check instead of executing it.
+#. Add your new malware check to the database by running ``docker-compose run web python -m warehouse malware sync-checks`` from the warehouse project root.
```

Comment:
Correct me if I'm wrong, but this probably requires an already initialized DB (i.e., `make initdb`). Might be worth noting before these steps that that's expected to have already been run.